### PR TITLE
More api

### PR DESCRIPTION
**Pull request by pentacular**

### CODE REVIEW

File: api/shape/Plan.js [hunk 1]
```diff
@@ -211,6 +211,52 @@ export const getScale = (geometry) => {
   ];
 };
 
+const X = 0;
+const Y = 1;
+const Z = 2;
+
+export const buildCorners = (x, y, z) => {
```
Comment (BarbourSmith) [inline api/shape/Plan.js:218]: This is a neat idea!

File: nb/api/Line.md [hunk 2]
```diff
@@ -41,11 +41,13 @@ Line(5)
 Line(5).and(rz(1 / 8)) produces a 1/8th angle.
 
 ```JavaScript
-Line(seq({ from: -1, upto: 1, by: 1 / 8 }, noOp, List))
+Line(seq({ from: -1, upto: 1, by: 1 / 8 }, noOp(), List))
```
Comment (BarbourSmith) [inline nb/api/Line.md:44]: Why the change to have noOp called instead of passed as a function? I think that the old way makes more sense to me intuitively

File: nb/api/Spiral.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
+### Spiral()
```
Comment (BarbourSmith) [inline nb/api/Spiral.md:1]: Can we do spirals with a z-component at the moment? Just curious

File: nb/api/SurfaceMesh.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
+### SurfaceMesh()
+Parameter|Default|Type
+---|---|---
+serialization||A serialized surface mesh.
+
+Reconstructs a surface mesh from its serialization.
+
+_This is intended for debugging._
```
Comment (BarbourSmith) [inline nb/api/SurfaceMesh.md:8]: 👍

File: nb/api/simplify.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
+### simplify()
+Parameter|Default|Type
+---|---|---
+eps||Number of mm for point resolution.
+{ratio}|1|Stop when the remaining edges reaches this fraction.
+
+_Check: does this actually do anything?_
```
Comment (BarbourSmith) [inline nb/api/simplify.md:7]: 😛

File: nb/api/size.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
+### size()
+Parameter|Default|Type
+---|---|---
+op|size => shape => size|Operates on the size.
+
+Analyzes the shape and calls op with an object with:
```
Comment (BarbourSmith) [inline nb/api/size.md:6]: Does .boundingBox() still exist? I kind of liked having the information returned like thisThingsSize = a.size();

File: nb/api/smooth.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+...selections||Only smooth where covered by a selection shape.
+{iterations}|1|
+{time}|1|
+{remeshIterations}|1|
+{remeshRelaxationSteps}|1|
+
+Smooths shape.
+
+```JavaScript
+Box(10, 10, 10)
```
Comment (BarbourSmith) [inline nb/api/smooth.md:14]: This is such a cool example

File: nb/api/sort.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
+### sort()
+Parameter|Default|Type
+---|---|---
+spec|'z<y<x<'|String defining the sort predicate.
+
+This orders a group of shapes by their extremal bounding box corner.
```
Comment (BarbourSmith) [inline nb/api/sort.md:6]: I didn't know about this one. That's awesome

File: nb/api/tint.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
+### tint()
+Parameter|Default|Type
+---|---|---
+color||The name of the color to tint with.
+
+Sets the color of all leaf geometry in shape to the named color, replacing existing colors.
+
+Mixes the named color with all leaf geometry, merging with existing colors.
```
Comment (BarbourSmith) [inline nb/api/tint.md:8]: This seems contradictory to me. Is it replacing the existing color or mixing with the existing color. It's mixing with it, right?

**Review by BarbourSmith — APPROVED** [review]
Loving all these descriptions. I'm learning things reading them so I think they will be super helpful to other people.

File: nb/api/size.md (hunk 6 above)
Comment (pentacular) [reply nb/api/size.md:6]: const foo = a.size(noOp()) should get you that, I think.

a.bb() exists and probably should have a long name like a.boundingBox()

File: nb/api/tint.md (hunk 9 above)
Comment (pentacular) [reply nb/api/tint.md:8]: Oops, yes.

File: nb/api/Line.md (hunk 2 above)
Comment (pentacular) [reply nb/api/Line.md:44]: noOp is a => b => b. So noOp() gets us the b => b which does what we want to pass through the numbers to gather with List.

But maybe this is a bit confusing -- might need to rethink it.

File: api/shape/Plan.js (hunk 1 above)
Comment (pentacular) [reply api/shape/Plan.js:218]: Yeah, should have factored that out earlier. :)

File: nb/api/Line.md (hunk 2 above)
Comment (BarbourSmith) [reply nb/api/Line.md:44]: Ah I understand now. It's a little confusing, but I don't think it's a big deal

File: nb/api/size.md (hunk 6 above)
Comment (BarbourSmith) [reply nb/api/size.md:6]: Nice, that works for me

File: nb/api/tint.md (hunk 9 above)
Comment (pentacular) [reply nb/api/tint.md:8]: Yes, you're right -- that first sentence needs deleting.
